pkg/api/server/audit/store/s3.go

```diff
@@ -0,0 +1,77 @@
+package store
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type S3StoreOptions struct {
+	AuditLogsStoreS3Bucket   string `usage:"Audit log store S3 bucket"`
+	AuditLogsStoreS3Endpoint string `usage:"Audit log store S3 endpoint"`
+}
+
+type s3Store struct {
+	host, bucket string
+	compress     bool
+	client       *s3.Client
+}
+
+func NewS3Store(host string, compress bool, options S3StoreOptions) (Store, error) {
+	if options.AuditLogsStoreS3Bucket == "" {
+		return nil, errors.New("Audit log store S3 bucket is required")
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Will this cause problems in situations where we're using an alternative S3 provider? Like if I had my own MinIO instance or something, would this still look for the AWS_* variables, and return an error if there are none?

Reply:
That's the way the workspace-provider works now, I believe, and this uses the same ideas.